### PR TITLE
Setup with TS

### PR DESCRIPTION
**Pull request by riteshsp2000**
Initialize Create React App with typescript
Setup Eslint and Prettier
Setup Testing with Jest and Enzyme

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -1,66 +1,84 @@
...
+  "name": "project-guava-web",
+  "description": "The Client Repository for Certificates Project",
+  "version": "0.1.0",
+  "private": true,
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/dscnitrourkela/project-guava.git"
+  },
+  "bugs": {
+    "url": "https://github.com/dscnitrourkela/project-guava.git"
```
Comment (DesignrKnight) [inline client/package.json:11]: guava-web in URL

File: client/package.json [hunk 2]
```diff
@@ -1,66 +1,84 @@
...
+      "not op_mini all"
+    ],
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
+  "dependencies": {
+    "jest": "^26.6.3",
```
Comment (DesignrKnight) [inline client/package.json:51]: do we need jest and webvitals in dependencies?
can be dev I suppose

File: client/package.json [hunk 3]
```diff
@@ -1,66 +1,84 @@
...
-    "babel-polyfill": "^6.26.0",
-    "babel-preset-react": "^6.24.1",
-    "clean-webpack-plugin": "^3.0.0",
-    "css-loader": "^4.3.0",
+    "@types/enzyme": "^3.10.8",
+    "@types/enzyme-adapter-react-16": "^1.0.6",
+    "@types/jest": "^26.0.20",
+    "@types/node": "^14.14.31",
+    "@types/react": "^17.0.2",
+    "@types/react-dom": "^17.0.1",
```
Comment (DesignrKnight) [inline client/package.json:64]: Don't we need these types on the server?
I suppose these are necessary for transcompiling on CI-CD/server

File: client/public/robots.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
+User-agent: *
+Disallow:
```
Comment (DesignrKnight) [inline client/public/robots.txt:3]: Add a TODO here, we need to have a proper ignore for robots.txt

File: client/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+  document.getElementById('root'),
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
```
Comment (DesignrKnight) [inline client/src/index.tsx:17]: Don't we need to have a check here to report only on dev and not on prod? and also the import also makes sense only in dev

Comment (DesignrKnight) [reply client/src/index.tsx:17]: if you are using env variables in package.json script. See dates for cross-env

File: client/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (DesignrKnight) [inline client/tsconfig.json:3]: We are targeting es5 here but prettier/eslint has es6 as standard
I hope they both do not clash

File: client/package.json (hunk 2 above)
Comment (riteshsp2000) [reply client/package.json:51]: I tried pushing it to dev, but react-scripts required jest as a dependency which is why moved it back from dev

File: client/package.json (hunk 3 above)
Comment (riteshsp2000) [reply client/package.json:64]: I guess they came as a dev dependency itself from CRA. Im not sure ill check and change accordingly.

File: client/public/robots.txt (hunk 4 above)
Comment (riteshsp2000) [reply client/public/robots.txt:3]: Okay noted

File: client/src/index.tsx (hunk 5 above)
Comment (riteshsp2000) [reply client/src/index.tsx:17]: This is all that came with CRA. No changes from my side. Ill look into it

File: client/tsconfig.json (hunk 6 above)
Comment (riteshsp2000) [reply client/tsconfig.json:3]: ill update it to es6... my bad

**Pull request closed without merging**